app/assets/javascripts/Components/Result/file_viewer.jsx

```diff
@@ -147,15 +147,15 @@ export class FileViewer extends React.Component {
         annotationFocus={this.props.annotationFocus}
         {...commonProps}
       />;
-    } else if (this.state.type !== '') {
+    } else if ((this.state.type !== '') && (this.state.type !== 'unknown')) {
```

Comment:
As discussed in the meeting. Only if a file is binary (not the same as having an unknown MIME type) the viewer should display the `'submissions.cannot_display'` message. We would like the same behaviour as when the file is viewed on the results page.

Please see `SubmissionsController.get_file` and `SubmissionFile.is_binary` for reference.

Reply:
Yes Misha, But we are setting the type to 'unknown' for binary files. 

submissions_controller.rb: line no: 444 to 446

I can add one more condition which checks for this type and display the 'submissions.cannot_display' message. And for the remaining type just return empty.

Why am I checking the unknown mime type in line no: 150 is, previously the Text Viewer including the binary files also.
So I eliminated that by checking the mime type should not be 'unknown' for text files.

Reply:
The `get_file` method does check that and sets the `type` to `'unknown'` but that method isn't being called here.  The content that is displayed is from the `download` method, not the `get_file` method. This is why the zip content is being displayed in the first place.